examples/C++/Keys/keys.cpp

```diff
@@ -35,25 +35,42 @@ typedef struct
     uint16_t max_samples_per_key;
 } example_configuration;
 
-class PubListener :public eprosima::fastrtps::PublisherListener
+class PubListener : public eprosima::fastrtps::PublisherListener
 {
 public:
-    PubListener() :n_matched(0), firstConnected(false) {};
+
+    PubListener()
+        : n_matched(0)
+        , firstConnected(false) {};
     ~PubListener() {};
-    void onPublicationMatched(eprosima::fastrtps::Publisher* /*pub*/, eprosima::fastrtps::rtps::MatchingInfo& /*info*/)
+    void onPublicationMatched(
+            eprosima::fastrtps::Publisher* /*pub*/,
+            eprosima::fastrtps::rtps::MatchingInfo& info)
     {
-        //runPublisher();
+        if (info.status == MATCHED_MATCHING)
+        {
+            n_matched++;
+        }
+        else
+        {
+            n_matched--;
+        }
     }
     int n_matched;
     bool firstConnected;
```

Comment:
I see #777 added `condition_variable`s whereas this PR doesn't. As a backport I would expect changes here to be exactly the same as in #777, what am I missing?